server/lib/manifest.py

```diff
@@ -0,0 +1,382 @@
+                                               level=AccessType.READ)
+                if folder:
+                    # Check if it's a dataset by checking for meta.identifier
+                    folder_meta = folder.get('meta')
+                    if folder_meta:
+                        dataset_identifier = folder_meta.get('identifier')
+                        if dataset_identifier:
+                            self.datasets.add(obj['itemId'])
+                            folder_files.append(
+                                {"dataset_identifier": dataset_identifier,
+                                 "provider": folder_meta.get('provider'),
+                                 "file_iterator": self.folderModel.fileList(folder,
+                                                                            user=self.user,
+                                                                            data=False)
+                                 })
+
+                    else:
+                        folder_files.append({"file_iterator": self.folderModel.fileList(
+                            folder,
+                            user=self.user,
+                            data=False)})
+            elif obj['_modelType'] == 'item':
+                """
+                If there is a file that was added to a tale that came from a dataset, but outside
+                the dataset folder, we need to get metadata about the parent folder and the file.
+
+                """
+                root_item = self.itemModel.load(obj['itemId'],
+                                                user=self.user,
+                                                level=AccessType.READ)
+                if root_item:
+                    # Should always be true since the item is in dataSet
+                    if root_item.get('meta'):
+                        item_folder = self.folderModel.load(root_item['folderId'],
+                                                            user=self.user,
+                                                            level=AccessType.READ)
+                        folder_meta = item_folder.get('meta')
+                        if folder_meta:
+                            self.datasets.add(root_item['folderId'])
+                            folder_files.append({"dataset_identifier":
+                                                folder_meta.get('identifier'),
+                                                 "provider":
+                                                     folder_meta.get('provider'),
+                                                 "file_iterator":
+                                                     self.itemModel.fileList(root_item,
+                                                                             user=self.user,
+                                                                             data=False)
+                                                 })
```

Comment:
There are items begin registered directly to root WholeTale Catalog. They have their own metadata set. Why folder metadata (in this case it's gonna be empty) takes precedence?

Reply:
Refactored to take into account items that are registered into the root catalogue.